component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/handler/DPoPAuthenticationHandler.java

```diff
@@ -0,0 +1,220 @@
+                    log.error(error, e);
+                }
+
+                String serviceProviderTenantDomain = null;
+                try {
+                    serviceProviderTenantDomain =
+                            OAuth2Util.getTenantDomainOfOauthApp(consumerKey);
+                } catch (InvalidOAuthClientException | IdentityOAuth2Exception e) {
+
+                    String error = String.format("Error occurred while getting the OAuth App" +
+                            " tenantDomain by Consumer key: %s", consumerKey);
+                    log.error(error, e);
+                }
+
+                if (serviceProvider != null) {
+                    authenticationContext.addParameter(DPoPConstants.SERVICE_PROVIDER, serviceProvider);
+                    if (serviceProviderTenantDomain != null) {
+                        authenticationContext.addParameter(DPoPConstants.SERVICE_PROVIDER_TENANT_DOMAIN,
+                                serviceProviderTenantDomain);
+                    }
+
+                    MDC.put(DPoPConstants.SERVICE_PROVIDER, serviceProvider);
+                    // Set OAuth service provider details to be consumed by the provisioning framework.
+                    setProvisioningServiceProviderThreadLocal(clientApplicationDTO.getConsumerKey(),
+                            serviceProviderTenantDomain);
+                }
+            }
+        }
+        return authenticationResult;
+    }
+
+    private AuthenticationResult getAuthenticationResult(AuthenticationResult authenticationResult,
+                                                         OAuth2TokenValidationResponseDTO responseDTO,
+                                                         String authorizationHeader,
+                                                         AuthenticationRequest authenticationRequest) {
+
+        if (!responseDTO.isValid()) {
+            return authenticationResult;
+        }
+
+        TokenBinding binding = responseDTO.getTokenBinding();
+        if (DPoPConstants.OAUTH_DPOP_HEADER.equals(binding.getBindingType())) {
+            if (!authorizationHeader.startsWith(DPoPConstants.OAUTH_DPOP_HEADER)) {
+                return authenticationResult;
+            }
+            String dpopHeader = authenticationRequest.getHeader(DPoPConstants.OAUTH_DPOP_HEADER);
+
+            if (StringUtils.isBlank(dpopHeader)) {
+                return authenticationResult;
+            }
+            try {
+                String thumbprintOfPublicKey = Utils.getThumbprintOfKeyFromDpopProof(dpopHeader);
+                if (StringUtils.isBlank(thumbprintOfPublicKey) ||
+                        !thumbprintOfPublicKey
+                                .equalsIgnoreCase(responseDTO.getTokenBinding().getBindingValue())) {
+                    return authenticationResult;
+                }
+            } catch (IdentityOAuth2Exception e) {
+                return authenticationResult;
```

Comment:
Why are we not throwing this?

Reply:
Fixed with [260a195](https://github.com/kalanika/identity-oauth-addons/commit/260a195559b47b115eac049a41977336d2c813b2).